planetary-social/nos
# Support naddr identifiers

## Issues covered
#1232 

## Description
This adds support for `naddr` identifiers in notes. With this PR, a user can write a note with a `nostr:naddr` link and we'll render it with our "Link to note", and tapping it will take you to the note.

This doesn't complete the above ticket; the note referenced in it contains Markdown `naddr` links and I haven't dug into that yet. Maybe it's easy, maybe not.

Update: I found out why the note from Onigirl doesn't open in Nos. Here's the link in the note:

`[Tap](nostr:naddr1qqyxvwfnxsmkxcfhqgsrexzf8qaaa2yrkz73dlkwrmfk6dlr0nw78njrk9l2f6ge9mq39zgrqsqqqa28kj8h7h)`

We don't currently support `nostr` as a URL scheme; we only support `nos` (or `nos-staging` for Staging, `nos-dev` for Dev). Should we add support for `nostr` as a URL scheme?

Update: I added support for `nostr` as a URL scheme.

## How to test
1. Navigate to `note1tyt3a7tqxhgvvve87jzxpsndguw6vq29qfuenau40kartujlrppq5zmncg`
2. Tap the Link to note
3. Observe

Expected: The note loads and shows the text.

Update: If you repeat the process (or just go back and tap the link again), the note does not load and shows and endless spinner. Need to fix this.

- [x] Fix endless spinner after tapping link, going back, then tapping link again

## Screenshots/Video
| Before | After |
|--------|--------|
| <img width="627" alt="Screenshot 2024-07-16 at 11 49 15 AM" src="https://github.com/user-attachments/assets/53663024-9cbb-42f3-abdc-665615e954b8"> | <img width="627" alt="Screenshot 2024-07-16 at 11 49 32 AM" src="https://github.com/user-attachments/assets/bc2c0989-b565-4498-bbad-5fe1bf03843a"> |
